src/core/providers/gdal/qgsgdalprovider.cpp

```diff
@@ -3008,9 +3008,14 @@ void QgsGdalProvider::initBaseDataset()
     }
     else
     {
+      // Add alpha band to the output VRT dataset so that pixels in empty regions
+      // (e.g. when the raster is rotated) will be transparent
+      gdal::warp_options_unique_ptr psWarpOptions( GDALCreateWarpOptions() );
+      psWarpOptions->nDstAlphaBand = GDALGetRasterCount( mGdalBaseDataset ) + 1;
```

Comment:
you should do something like (untested, you'll need to test with a RGBA input image):
```
if( GDALGetRasterColorInterpretation(GDALGetRasterBand(mGdalBaseDataset, GDALGetRasterCount( mGdalBaseDataset ))) == GCI_AlphaBand )
{
    psWarpOptions->nBandCount = GDALGetRasterCount( mGdalBaseDataset ) - 1;
    psWarpOptions->nDstAlphaBand = GDALGetRasterCount( mGdalBaseDataset );
}
else
   psWarpOptions->nDstAlphaBand = GDALGetRasterCount( mGdalBaseDataset ) + 1;
```

And this logic should be moved before the ``if ( GDALGetMetadata( mGdalBaseDataset, "RPC" ) )`` line so that psWarpOptions is passed to QgsGdalUtils::rpcAwareAutoCreateWarpedVrt() too

Reply:
Thanks Even! Indeed for RGBA rasters my code was adding another alpha band.

In your code, I am not sure if I understand the logic behind setting `nBandCount` - and that line is causing a crash in GDALAutoCreateWarpedVRT(). When I remove the line with `nBandCount`, things work well. And when testing the code, I have realized that for RGBA rasters I don't even need to set `nDstAlphaBand` to get things to work, it seems to be picked up automatically (see 50060e2)